pkg/vsphere/disk/disk_manager.go

```diff
@@ -469,7 +469,7 @@ func (m *Manager) detach(op trace.Operation, disk *types.VirtualDisk) error {
 		return t, er
 	})
 
-	if err != nil {
+	if err == nil {
```

Comment:
@chengwang86 It's a channel - there's no way to tell how many items are in it. It's the main curse of using one as a counting semaphore, but they are very simple and well tested.